test/Jsonb.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-unused-imports -Wno-orphans -Wno-name-shadowing #-}
+
+module Jsonb (specJsonb) where
+
+import Control.Monad.Except
+import Control.Monad.Identity
+import Control.Monad.Reader
+import Data.Aeson qualified as J
+import Data.ByteString qualified as BS
+import Data.Kind (Type)
+import Data.String
+import Database.PG.Query
+import Database.PG.Query.Connection
+import Database.PostgreSQL.LibPQ.Internal
+import GHC.Generics
+import System.Environment qualified as Env
+import Test.Hspec
+import Prelude
+
+type TestValue :: Type
+newtype TestValue = TestValue {hey :: Int}
+  deriving stock (Show, Generic)
+
+instance J.FromJSON TestValue
+
+getPgUri :: (MonadIO m) => m BS.ByteString
+getPgUri = liftIO $ fromString <$> Env.getEnv "DATABASE_URL"
+
+getPostgresConnect :: (MonadIO m) => m ConnInfo
+getPostgresConnect = do
+  dbUri <- getPgUri
+  pure $
+    defaultConnInfo
+      { ciDetails = CDDatabaseURI dbUri
+      }
+
+specJsonb :: Spec
+specJsonb = do
+  describe "Decoding JSON and JSONB" $ do
+    it "Querying 'json' from PostgreSQL" $ do
+      pg <- getPostgresConnect
+      SingleRow (Identity (i :: BS.ByteString)) <-
+        runTxT
+          pg
+          (rawQE show "select '{\"hey\":42}'::json" [] False)
+      i `shouldSatisfy` const True
```

Comment:
That the SQL query does not throw an error. Could be clearer though, will make it return an `Either` and do `isRight` instead.

Reply:
Yeah maybe a comment describing what we are trying to test would be helpful as well, otherwise someone might mistake this as a test that was not completed.